python/lsst/ctrl/bps/htcondor/htcondor_service.py

```diff
@@ -1392,6 +1408,36 @@ def _get_run_summary(job):
     return summary
 
 
+def _get_exit_code_summary(jobs):
+    """Get the exit code summary for a run.
+
+    Parameters
+    ----------
+    jobs : `dict` [`str`, `dict` [`str`, Any]]
+        Mapping HTCondor job id to job information.
+
+    Returns
+    -------
+    summary : `dict` [`str`, `list` [`int`]]
+        Jobs' exit codes per job label.
+    """
+    summary = {}
+    for job_id, job_ad in jobs.items():
+        job_label = job_ad["bps_job_label"]
+        summary.setdefault(job_label, [])
+        try:
+            if job_ad["JobStatus"] in {JobStatus.COMPLETED, JobStatus.HELD}:
+                try:
+                    exit_code = job_ad["ExitSignal"] if job_ad["ExitBySignal"] else job_ad["ExitCode"]
+                except KeyError:
+                    _LOG.debug("Cannot determine exit code for job '%s'", job_id)
```

Comment:
Is there ever a case when ExitBySignal may not be in dictionary but ExitCode will be?  If so, this code needs to be updated to allow for that case.

Reply:
I wasn't able to find a definitive answer for your question in the HTCondor docs, but based on the tests I made and HTCondor logs I looked at, it seems HTCondor *always* sets `ExitBySignal` in a job ClassAd (and `ExitSignal` and `ExitCode` are set conditionally depending what the value of the `ExitBySignal` is).